File: tests/Movies.IntegrationTests/ApiResponse.cs
namespace Movies.IntegrationTests;

public class ApiResponse
{
    public bool Ok { get; set; }
    public string? Code { get; set; }
    public string[]? Details { get; set; }
}

public class ApiResponse<TValue> : ApiResponse
{
    public TValue? Value { get; set; }
}

File: frontend/types/ApiResponse.ts
interface IResponse {
  ok: boolean;
}

interface IErrorResponse {
  ok: false;
  code: string;
}

interface ISuccessResponse<TValue> {
  ok: true;
  value: TValue;
}

type ApiResponse<TValue> = IErrorResponse | ISuccessResponse<TValue>;

export default ApiResponse;
